packages/ripple-keypairs/package.json

```diff
@@ -17,10 +17,8 @@
     "test": "test"
   },
   "dependencies": {
-    "bn.js": "^5.1.1",
-    "brorand": "^1.0.5",
-    "elliptic": "^6.5.4",
-    "hash.js": "^1.0.3",
+    "@noble/curves": "^1.0.0",
+    "@xrpl/crypto": "1.0.0",
```

Comment:
regarding @xrpl/crypto, it was ripple-iso-crypto, and before that it was ripple-hashes. It was mostly a tree-shakeable way of getting isomorphic hashes, using native crypto where possible (primarily for performance). The @noble author didn't want to add any isomorphic code to his library. I did have in mind something generic, that could potentially be used outside of ripple. But we could go the route you seem to be thinking of, where the crypto is abstracted away. You could use native libs on node for a lot of the secp256k1/ed25519 operations. It's maybe overkill though?

I had pondered down that line before: https://github.com/XRPLF/xrpl.js/issues/2306#issuecomment-1547129006

(re: the Group interface, after some recent experience patching @cloudflare/voprf-ts, I believe you'd want to have a Scalar object, instead of using bigint directly) 

So, we could go a few ways, just use @noble/hashes directly (perf probably "good enough" for large majority of purposes), and kill off @xrpl/crypto, or go full tilt in that direction with the abstractions (and potential perf enhancements) ... 

Which way do you all want to head ?

Reply:
Native/performant add-on forsecp256k1 
https://www.npmjs.com/package/secp256k1

Reply:
Chat says:

Your proposal sounds reasonable. Focusing on critical features first and building a minimalistic library can have several advantages:

1. **Scoping Down:** By keeping the scope of @xrpl/crypto to hashing primitives, randomBytes utils, sha512half, and similar utilities, you can focus on delivering high-quality code that directly addresses your project's needs.

2. **Abstraction Later:** Delaying abstraction of the ECC curve operations allows you to better understand your needs and only develop those abstractions that provide tangible value. This aligns with the YAGNI (You Aren't Gonna Need It) principle in software development, which encourages not to add functionality until deemed necessary.

3. **Switch to Active Libraries:** Replacing the deprecated elliptic library with the more actively maintained @noble libraries sounds like a good move to ensure your project stays up-to-date and receives important updates and security patches.

4. **Performance:** If you find that performance is a bottleneck in your Node environment later on, you can consider adding support for native add-ons like node-secp256k1. By then, you'll have a better idea of your project's performance requirements and how much improvement these native add-ons can provide.

Remember to perform thorough testing as you make these transitions, to ensure your code continues to behave as expected. It can be helpful to have a set of benchmark tests, to compare performance before and after the changes. If the changes are made progressively and tested thoroughly at each stage, you should be able to execute this plan with minimal risk.